src/kudu/consensus/consensus_meta.h

```diff
@@ -225,6 +241,13 @@ class ConsensusMetadata : public RefCountedThreadSafe<ConsensusMetadata> {
   DFAKE_MUTEX(fake_lock_);
 
   std::string leader_uuid_; // Leader of the current term (term == pb_.current_term).
+
+  // Flexible quorums artifacts.
+  LastKnownLeader last_known_leader_;  // Last known leader
+
+  // Previous granted vote
+  std::map<int64_t, PreviousVotePB> previous_vote_history_;
```

Comment:
previous_vote_history_ is not part of ConsensusMetaPB, which means that we will lose the history of votes if an instance restarts. Is this going to impact the algorithm?

Reply:
Yes. I plan to add it to ConsensusMetaPB. Let me add a TODO